plugins/extpoints/interfaces.go

```diff
@@ -11,13 +12,14 @@ import (
 //
 // We wrap all arguments so that we can add additional properties without
 // breaking source compatibility with older plugins.
+// Note: This is passed by-value for efficiency (and to prohibit nil), if
+// adding any large fields please consider adding them as pointers.
+// Note: This is intended to be a simple argument wrapper, do not add methods
+// to this struct.
 type PluginOptions struct {
 	environment *runtime.Environment
 	engine      *engines.Engine
-	// Note: This is passed by-value for efficiency (and to prohibit nil), if
-	// adding any large fields please consider adding them as pointers.
-	// Note: This is intended to be a simple argument wrapper, do not add methods
-	// to this struct.
+	log         *logrus.Entry
```

Comment:
So if we did the interface thing... We could to just hide the `WithFields` method as it requires `logrus.Fields`... Looking at this code I think `WithField(string, ínterface{})` is sufficient... In rare cases where we have two fields we can just do `log.WithField(...).WithField(...)`..